Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -1236,10 +1405,30 @@ namespace Terrain
             }
         }
 
+        if (m_detailTextureImage && m_areaData.m_heightmapImage && !m_imagesSetOnSrgs)
+        {
+            m_imagesSetOnSrgs = true;
+            for (auto& view : process.m_views)
+            {
+                auto viewSrg = view->GetShaderResourceGroup();
+                viewSrg->SetImage(m_heightmapPropertyIndex, m_areaData.m_heightmapImage);
```

Comment:
This is an area that's likely to change a lot in the future, so it's easiest to keep here for now. I didn't put it in the scene srg because the heightfield may eventually be different depending on where the camera is (imagine two cameras very far apart where do don't want all the height data between them resident). I also didn't put it in the TerrainSrg because then I'd have to bind the TerrainSrg to the shadow passes and depth prepass. In the short term we may just end up baking the heights into the meshes themselves so they play nicely with ray tracing, at least until we can add a hit shader for ray tracing that takes a heightmap into account.